### DISCUSSION TITLE
Adaptive Bitrate strategies

### DISCUSSION THREAD

**Discussion by fernandomaura97** [alvr-org/ALVR#1859] [Q&A]
Hi all, I'm doing some active research on ABR algorithms as part of my phd; I've been working on implementing some low overhead strategies for ABR and have been working on implementing them on C++ and now rust on a recent fork of ALVR. 
Although this is still yet an early work in progress, I've wrapped my head around creating new settings to add to the dashboard in order to initialize my library which I show [here](https://www.diffchecker.com/w3E8UEd4/). 

In the near future I'm going to work on trying out different strategies ( i.e. RL-based, BOLA, Google Congestion Control-like)  but this is currently in an early stage and still needs to be tested on the hardware; currently I would really appreciate any pointers as for the recommended way to interact with the codebase in a safe way for the rest of the pipeline.

The only problem I have now is that I've been having some trouble wrapping my head around **how to access/borrow into my library these variables that the user interacts with from the dashboard**. My library file is located in a placeholder atm in [/alvr/aa_algorithms/.](https://github.com/fernandomaura97/ALVR-adaptive/tree/master/alvr/aa_algorithms). 
Thank you!

**Comment by xytovl** [comment]
Usually the easy way is to look at how it is done for the rest of the code, for instance follow `max_bitrate_mbps` from existing bitrate code and it will show you where it is used, initialized and interacts with the rest of the code.

Development discussions are mostly over discord (see link in the readme), you will get more answers there.
